File: src/main/java/com/example/campusoverflow/question/Question.java
package com.example.campusoverflow.question;

import com.example.campusoverflow.common.BaseEntity;
import com.example.campusoverflow.tag.Tag;
import com.example.campusoverflow.user.User;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.util.List;

@Getter
@Setter
@ToString(callSuper = true)
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Question extends BaseEntity {

    private String title;
    private String content;
    @ManyToMany(cascade = CascadeType.PERSIST, fetch = FetchType.EAGER)
    @JoinTable(
            name = "question_tag",
            joinColumns = @JoinColumn(name = "question_id"),
            inverseJoinColumns = @JoinColumn(name = "tag_id")
    )
    private List<Tag> tags;
    @ManyToOne(cascade = CascadeType.PERSIST, fetch = FetchType.EAGER)
    @JoinColumn(nullable = false)
    private User user;
}

File: CampusOverflow-Front/src/app/services/models/question.ts
/* tslint:disable */
/* eslint-disable */
import { Tag } from '../models/tag';
import { User } from '../models/user';
export interface Question {
  content?: string;
  createdAt?: string;
  createdBy?: number;
  id?: number;
  tags?: Array<Tag>;
  title?: string;
  updatedAt?: string;
  updatedBy?: number;
  user?: User;
}
